### ISSUE TITLE
Bunch of values for scores

### ISSUE THREAD

**Issue by ImGere** [dotnet/machinelearning-modelbuilder#126]
**System Information (please complete the following information):**
 - Model Builder Version: ML.NET 1.1.0
 - Visual Studio Version: 2019

**Describe the bug**
Dunno if it is a bug, but when i try to run a model, i get back some strange results. Can't find some documentation about it.

**To Reproduce**
1. Create model with model builder
2. In 2500 secs, it gave me back only one trainer with 39%. The dataset is around 1MB with 70k rows.
3. rerun the model after creating the project

**Screenshots**
![Image](https://user-images.githubusercontent.com/20186579/60197083-01ad3180-983f-11e9-8db4-cdfda25b5685.png)

**Comment by justinormont** [comment]
@ImGere: You may be running as multiclass and the one labeled displayed looks like you may instead want to use regression.

**Comment by ImGere** [comment]
No i need multiclass, because there are codes, like 4200 are ids, don't need regression. I need my model to identify the correct ID. Don't know if I explained myself very well :)

**Comment by justinormont** [comment]
How many label values do you have?

**Comment by ImGere** [comment]
The thing I don't understand is what these floats means. Are these the probabilities of a specific class?
Also how can I get the test set in order to not make prediction with the data that's been used for training?
Another question is, how can I recreate the model after the model builder has done it. If my data changes I want to be able to recreate the model in order to fit new data. How can i do it without using the model builder? I tried to run the ModelBuilder.cs class but after 30 mins it was stuck on Evaluate

**Comment by justinormont** [comment]
These are the probabilities of a specific class.

The AutoML will automatically split your dataset into a Training and Validation set. So assuming no duplicate rows are present, and it's ok to split the rows randomly between train & validate, there will be no leakage between the datasets.

I'm uncertain where it would be stuck. Are you able to post your dataset here (publicly)?

**Comment by ImGere** [comment]
I can't post my dataset here because it's private data.

I don't know how to improve my model without being able to recreate the it. And 80 mins for the evaluation are maybe too much.

> For choosing regression vs. multiclass-classification:
> Is there an order to your classes? Eg, is class 1031 more similar to 1032 than class 3000? If so, you may want to run as a regression task, then round the predicted Score to the closest Integer (or nearest class).

No, as I said before it's not regression, classes are random number, just ids.

> The printed floats are the probabilities of the sample belonging to each class.

 Is there a way to retrieve which class as this probability? Because I need to show two or three answers of my model and I would like it to be the classes with the highest probability.

**Comment by ImGere** [comment]
I have found a way to implement the probability of every class, but there seems to be an error inside the library, can you check please?
[I opened an issue on where i found it](https://github.com/dotnet/machinelearning-samples/issues/542)

**Comment by ImGere** [comment]
Any news on the multiple predictions or an easy way to get the classes of the model?

Another question I have is about the documentation, I was trying to find documentation about trainers and evaluators, but I found very little.

**Comment by greazer** [comment]
ImGere, there seems to be several questions in here. One or two related to Model Builder, one to the sample you're using, and one related to how to process multiclass training output to get the information you need and one on documentation.

As for retraining on model builder, there is currently no way to retrain a model based on the training you've done before. This is something we plan on adding eventually.

As for an easier way to process output for getting the classes for the model, it sounds as though you were able to make some progress here on your own. We don't currently have any other answer for this question. 

For the documentation question, what have you found so we can compare to what is currently available?

**Comment by ImGere** [comment]
Hi @greazer thanks for the answer.

> One or two related to Model Builder, one to the sample you're using, and one related to how to 
> process multiclass training output to get the information you need and one on documentation.

For the questions regarding documentation and multiclass training, i'll explain a bit more, I worked on this project initially in python, where I found out that decision trees and Random forest could help me solve this problem pretty easily and it worked. But when ML.NET came out with an official release I decided to give it a try and didn't go so well as planned since the documentation initially was pretty low, now I did a quick research and it got better. But, for example, in [trainers](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.trainers?view=ml-dotnet) they are just listed here in not a particular order, could be useful to divide them into regression/classification, or some major areas.

> As for an easier way to process output for getting the classes for the model, it sounds as though you 
> were able to make some progress here on your own. We don't currently have any other answer for this 
> question.

As for this, I think it could be useful. I found a way to resolve this in [#542](https://github.com/dotnet/machinelearning-samples/issues/542) but it is still unsolve.
I could try to solve it on my own but docs are not that helpful: 
[link](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.predictionenginebase-2.outputschema?view=ml-dotnet#Microsoft_ML_PredictionEngineBase_2_OutputSchema)

Thank you for your time, but I keep on asking since I want to help you improve this library :D
